rotkehlchen/chain/evm/decoding/gearbox/gearbox_cache.py

```diff
@@ -0,0 +1,327 @@
+import logging
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Literal, Optional
+
+from rotkehlchen.assets.asset import UnderlyingToken
+from rotkehlchen.assets.utils import TokenEncounterInfo, get_or_create_evm_token
+from rotkehlchen.chain.evm.constants import ETH_SPECIAL_ADDRESS
+from rotkehlchen.chain.evm.contracts import EvmContract
+from rotkehlchen.chain.evm.decoding.gearbox.constants import GEARBOX_DATA_COMPRESSOR
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.constants.misc import ONE, ZERO
+from rotkehlchen.errors.misc import NotERC20Conformant, RemoteError
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.globaldb.cache import (
+    globaldb_get_general_cache_keys_and_values_like,
+    globaldb_get_general_cache_values,
+    globaldb_get_unique_cache_value,
+    globaldb_set_general_cache_values,
+    globaldb_set_unique_cache_value,
+)
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_evm_address
+from rotkehlchen.types import CacheType, ChecksumEvmAddress, EvmTokenKind
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.db.dbhandler import DBHandler
+    from rotkehlchen.db.drivers.gevent import DBCursor
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+@dataclass(init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=False)
+class GearboxPoolData:
+    """Mapping of gearbox pool rewards address to underlying pool name"""
+    pool_address: ChecksumEvmAddress
+    pool_name: str
+    farming_pool_addresses: set[str]
+    lp_token_addresses: set[str]
+    underlying_token: ChecksumEvmAddress
+
+
+def get_existing_pools(
+        cursor: 'DBCursor',
+        cache_type: Literal[CacheType.GEARBOX_POOL_ADDRESS],
+) -> set[ChecksumEvmAddress]:
+    """Returns all the gearbox pool rewards address stored in cache"""
+    return {
+        string_to_evm_address(address) for address in globaldb_get_general_cache_values(
+            cursor=cursor,
+            key_parts=(cache_type,),
```

Comment:
you will add in a different PR arbitrum one. If you leave the cache type like this it will you will be pullin arb and ethereum pools mixed. The key should have an extra part that is the chain_id (the integer)